src/utils.jl

```diff
@@ -95,31 +95,14 @@ function to_extent(edges::Vector{Edge})
     Extents.Extent(X=x, Y=y)
 end
 
-function to_points(x, ::Type{T} = Float64) where T
-    points = Vector{TuplePoint{T}}(undef, _npoint(x))
-    _to_points!(points, x, 1)
-    return points
-end
+_to_point3(T, p) = (T(GI.x(p)), T(GI.y(p)), T(GI.z(p)))
+_to_point2(T, p) = (T(GI.x(p)), T(GI.y(p)))
 
-_to_points!(points::Vector, x, n) = _to_points!(points, trait(x), x, n)
-function _to_points!(points::Vector, ::FeatureCollectionTrait, fc, n)
-    for f in GI.getfeature(fc)
-        n = _to_points!(points, f, n)
-    end
-end
-_to_points!(points::Vector, ::FeatureTrait, f, n) = _to_points!(points, GI.geometry(f), n)
-function _to_points!(points::Vector, ::AbstractGeometryTrait, fc, n)
-    for f in GI.getgeom(fc)
-        n = _to_points!(points, f, n)
-    end
-end
-function _to_points!(points::Vector, ::Union{AbstractCurveTrait,MultiPointTrait}, geom, n)
-    n = 0
-    for p in GI.getpoint(geom)
-        n += 1
-        points[n] = _tuple_point(p)
-    end
-    return n
+"""
+    to_points(geom::AbstractGeometry, ::Type{T} = Float64) where {T <: Number}
+"""
+function to_points(geom, ::Type{T} = Float64) where {T <: Number}
+    return collect(flatten(Base.Fix1(_is3d(geom) ? _to_point3 : _to_point2, T), GI.PointTrait, geom))
```

Comment:
Deleted the others as I couldn't do a suggestion:
```suggestion
    if is3d(geom)
        return collect(flatten(p -> _to_point3(T, p)), GI.PointTrait, geom))
    else
        return collect(flatten(p -> _to_point2(T, p), GI.PointTrait, geom))
    end
```

This way `T` is type stable and and its more readable. Puttting a type in a Fix2 or as a closure argument doesn't work, its just `DataType`.

Reply:
Makes sense.  Thanks for the suggestion!

I didn't realize that the Fix2 doesn't propagate - that's a whoops.  It seemed to improve runtime earlier but that may have been a purely compile-time improvement.